### ISSUE TITLE
finetuning with batch size > 1

### ISSUE THREAD

**Issue by mio2333** [kohya-ss/sd-scripts#284]
I tried to finetune a model based on [AOM3A1](https://huggingface.co/WarriorMama777/OrangeMixs#aom3a1) with 791 images.

It worked when I set the batch size is 1, but with batch size > 1,then it failed as follows:
steps:   0%|                                                                                                                                                                      | 0/31200 [00:00<?, ?it/s
epoch 1/300                                                                                                                                                                                                steps:   0%|▍                                                                                                                                             | 104/31200 [01:22<6:50:07,  1.26it/s, loss=0.139
epoch 2/300                                                                                                                                                                                                 steps:   1%|▉                                                                                                                                             | 208/31200 [02:42<6:43:39,  1.28it/s, loss=0.138
epoch 3/300                                                                                                                                                                                                 steps:   1%|█▍                                                                                                                                            | 312/31200 [04:02<6:39:54,  1.29it/s, loss=0.141
epoch 4/300 
steps:   1%|█▉                                                                                                                                            | 416/31200 [05:22<6:37:51,  1.29it/s, loss=0.141
epoch 5/300
steps:   2%|██▎                                                                                                                                           | 500/31200 [06:27<6:36:03,  1.29it/s, loss=0.136
steps:   1%|▉                                                                                                                                             | 208/31200 [02:42<6:43:39,  1.28it/s, loss=0.138]
epoch 3/300
steps:   1%|█▍                                                                                                                                            | 312/31200 [04:02<6:39:54,  1.29it/s, loss=0.141]
epoch 4/300
steps:   1%|█▉                                                                                                                                            | 416/31200 [05:22<6:37:51,  1.29it/s, loss=0.141]
epoch 5/300
steps:   2%|██▎                                                                                                                                           | 500/31200 [06:27<6:36:03,  1.29it/s, loss=0.136]
steps:   2%|██▎                                                                                                                                           | 520/31200 [06:51<6:44:26,  1.26it/s, loss=0.137]
epoch 6/300
steps:   2%|██▊                                                                                                                                           | 624/31200 [08:10<6:40:37,  1.27it/s, loss=0.146]
epoch 7/300
steps:   2%|███▎                                                                                                                                          | 728/31200 [09:29<6:37:36,  1.28it/s, loss=0.138]
epoch 8/300
steps:   3%|███▋                                                                                                                                            | 810/31200 [10:32<6:35:16,  1.28it/s, loss=nan]

This loss looks normal before it becomes nan.

This is my training command:
python -u /path/project/stable-diffusion-webui/extensions/kohya-sd-scripts-webui/kohya_ss/fine_tune.py --pretrained_model_name_or_path /path/project/stable-diffusion-webui/models/Stable-diffusion/AOM3A1.safetensors --train_data_dir /path/train_data_dir --caption_extension .txt  --enable_bucket --min_bucket_reso 256 --max_bucket_reso 1024 --dataset_repeats 1 -- output_dir \                                                                                                                                                                                                /path/project/stable-diffusion-webui/models/Stable-diffusion/ --output_name minmix --save_precision fp16 --save_n_epoch_ratio 6 --train_batch_size 8 --max_token_length 225 --xformers --               max_train_epochs 300 --max_data_loader_n_workers 8 --persistent_data_loader_workers --gradient_accumulation_steps 1 --mixed_precision fp16 --clip_skip 2 --logging_dir /path/download/                  train_work_dir/  --sample_every_n_steps 500 --sample_prompts \                                                                                                                                              /path/train_work_dir/sample_prompts.txt --sample_sampler euler_a --save_model_as safetensors --optimizer_type AdamW8bit --learning_rate 2e-06 --max_grad_norm 1.0 --lr_scheduler constant --   lr_scheduler_num_cycles 1 --lr_scheduler_power 1.0 --in_json /path/train_data_dir/meta.json --resolution 512,512      

I test all the situations when batch size is equal to 2,4, and 8. All fail like above, get nan loss and the sample image is black.
I read the finetune code and can not find the reason.
Appreciate it if anyone can help, thanks!
